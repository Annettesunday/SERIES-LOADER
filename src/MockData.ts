export const episodes = [
  {
    airdate: "2013-12-02",
    airstamp: "2013-12-03T03:30:00+00:00",
    airtime: "22:30",
    id: 14308,
    image: {
      medium:
        "http://static.tvmaze.com/uploads/images/medium_landscape/15/37912.jpg",
      original:
        "http://static.tvmaze.com/uploads/images/original_untouched/15/37912.jpg"
    },
    name: "Pilot",
    number: 1,
    runtime: 30,
    season: 1,
    summary:
      "<p>Rick takes Morty to another dimension to get some seeds for him but Morty's parents are considering to put Rick in a retirement home for keeping Morty away from school to help him in his lab.</p>",
    url: "http://www.tvmaze.com/episodes/14308/rick-and-morty-1x01-pilot"
  }
];
