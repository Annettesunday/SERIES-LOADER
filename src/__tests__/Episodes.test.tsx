import React from "react";
import {
  render,
  cleanup,
  fireEvent,
  waitForElement,
  getByText
} from "@testing-library/react";
import EpisodesList from "../EpisodesList";
import { IEpisodeProps } from "../interfaces";
import { episodes } from "../MockData";

afterEach(cleanup);

describe("<EpisodesList />", () => {
  function renderEpisodesComponent(props: Partial<IEpisodeProps> = {}) {
    const defaultProps = {
      episodes,
      store: {
        state: {
          episodes,
          favorites: []
        },
        dispatch: jest.fn()
      },
      toggleFavAction: jest.fn(),
      favorites: []
    };
    return render(<EpisodesList {...defaultProps} {...props} />);
  }
  test("matches snapshot", () => {
    const { asFragment } = renderEpisodesComponent();
    expect(asFragment()).toMatchSnapshot();
  });
  test("it displays the episode provided via props", async () => {
    const { getByTestId, queryByTestId } = renderEpisodesComponent();
    const episodeName = await getByTestId(/episode-name/i);
    const seasonDetails = await getByTestId(/season-number-test/i);
    const episodeContainer = await queryByTestId(/episode-box/i);

    expect(episodeContainer).toBeTruthy();
    expect(episodeName.textContent).toBe("Pilot");
    expect(seasonDetails.textContent).toBe("Season: 1 Number: 1");
  });

  test("it should display a button with the text fave on initial render", async () => {
    const { getByTestId } = renderEpisodesComponent();

    const faveButton = await getByTestId(/button-test/i);
    expect(faveButton.textContent).toBe("Fave");
  });

  test("toggleFavAction method should be called when fave button is clicked", async () => {
    const toggleFavAction = jest.fn();
    const { getByTestId } = renderEpisodesComponent({
      toggleFavAction
    });

    const faveButton = await getByTestId("button-test");
    fireEvent.click(faveButton);
    expect(toggleFavAction).toHaveBeenCalled();
  });
});
