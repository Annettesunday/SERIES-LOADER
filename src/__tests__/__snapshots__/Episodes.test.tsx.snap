// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EpisodesList /> matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="episode-box"
    data-testid="episode-box"
  >
    <img
      alt="Rick and Morty Pilot"
      src="http://static.tvmaze.com/uploads/images/medium_landscape/15/37912.jpg"
    />
    <div
      data-testid="episode-name"
    >
      Pilot
    </div>
    <section
      style="display: flex; justify-content: space-between;"
    >
      <div
        data-testid="season-number-test"
      >
        Season: 1 Number: 1
      </div>
      <button
        data-testid="button-test"
        type="button"
      >
        Fave
      </button>
    </section>
  </section>
</DocumentFragment>
`;
